### ISSUE TITLE
CI test flake: waiting to discover ClickHouse port

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/omicron#5180]
I was surprised not to find another test flake issue citing the same error message.

https://github.com/oxidecomputer/omicron/runs/22187324164

```
--- STDERR:              omicron-nexus::test_all integration_tests::disks::test_disk_region_creation_failure ---
 file: /var/tmp/omicron_tmp/test_all-f286e8141516cb30-test_disk_region_creation_failure.51137.0.log
 configured to log to "/var/tmp/omicron_tmp/test_all-f286e8141516cb30-test_disk_region_creation_failure.51137.0.log"
 URL: postgresql://root@[::1]:44403/omicron?sslmode=disable
 address: [::1]:44403
 'integration_tests::disks::test_disk_region_creation_failure' panicked at /work/oxidecomputer/omicron/nexus/test-utils/src/lib.rs:395:10:
 `Result::unwrap()` on an `Err` value: waiting to discover ClickHouse port
 by:
    non-retryable error while polling on condition: Invalid ClickHouse port number

```

**Comment by bnaecker** [comment]
I have a theory for how this happened. We search the logfile for the port by finding something that looks like `http://[<IP>]:<PORT>`, and then split the line on that last `:` and parse the number that follows. If we just so happen to get a line that ends directly at the colon, we'll get this error when we try to parse an empty string.

I've written a reproducer for this, and can put up a PR later this morning.

**Comment by bnaecker** [comment]
This should be resolved by #6655.
